docs/storage-architecture.md

```diff
@@ -0,0 +1,189 @@
+# Media Files Storage Architecture
+
+## Overview
+
+The media files system is organized into three distinct layers, each with a specific responsibility:
+
+```
+┌─────────────────────────────────────────────────────────────┐
+│                    whitenoise/media_files                    │
+│                   (High-level orchestration)                 │
+│  - Coordinates between storage, database, and Blossom       │
+│  - Handles encryption/decryption                            │
+│  - Manages business logic                                   │
+└────────────┬──────────────┬──────────────┬──────────────────┘
+             │              │              │
+             ▼              ▼              ▼
+    ┌────────────┐  ┌──────────────┐  ┌──────────────┐
+    │     DB     │  │      FS      │  │   Blossom    │
+    └────────────┘  └──────────────┘  └──────────────┘
+    database/       storage/          BlossomClient
+    media_files     media_files       (nostr_sdk)
+```
+
+## Layer Responsibilities
+
+### 1. `whitenoise/media_files.rs` (Orchestration Layer)
+**Purpose:** High-level coordination between storage and database
+
+**Responsibilities:**
+- Coordinate between storage layer and database layer
+- Provide convenience methods for common operations
+- Combine filesystem and database operations atomically
+- Business logic and validation
+
+**Does NOT:**
+- Touch the filesystem directly (delegates to storage layer)
+- Execute SQL directly (delegates to database layer)
+- Handle network operations (use BlossomClient)
+- Handle encryption/decryption (MDK's responsibility)
```

Comment:
Maybe clarify "Handle encryption/decryption **directly**", as the above diagram says:

```
┌─────────────────────────────────────────────────────────────┐
│                    whitenoise/media_files                   │
│                   (High-level orchestration)                │
│  - Coordinates between storage, database, and Blossom       │
│  - Handles encryption/decryption                            │
│  - Manages business logic                                   │
└────────────┬──────────────┬──────────────┬──────────────────┘
```

so it seems a bit contradictory or at least it can be confusing